gptel-bedrock.el

```diff
@@ -513,13 +513,34 @@ Non-nil CLEAR-CACHE will refresh credentials."
        (gptel-bedrock--fetch-aws-profile-credentials profile t))
       (t (user-error "AWS credentials expired for profile: %s" profile)))))
 
+(defun gptel-bedrock--fetch-instance-credentials ()
+  "Fetch AWS credentials using aws configure export-credentials.
+
+Returns a list of access key, secret key, and session token."
+  (with-temp-buffer
+    (condition-case nil
+        (progn
+          (call-process "aws" nil t nil "configure" "export-credentials")
+          (unless (zerop (buffer-size))
+            (let* ((creds (json-parse-string (buffer-string) :object-type 'plist))
+                   (access-key (plist-get creds :AccessKeyId))
+                   (secret-key (plist-get creds :SecretAccessKey))
+                   (session-token (plist-get creds :SessionToken)))
+              (cl-values access-key secret-key session-token))))
+      (error nil))))
```

Comment:
The error message must be more informative, depending on whether `aws` was (i) available, (ii) called and (iii) failed to return a token.

Currently the request fails silently with no message, since you've condition-cased the `call-process`.